src/Sylius/Bundle/ApiBundle/spec/CommandHandler/Cart/ApplyCouponToCartHandlerSpec.php

```diff
@@ -84,4 +84,19 @@ function it_throws_exception_if_promotion_coupon_is_not_found(
         $this->shouldThrow(\InvalidArgumentException::class)
             ->during('__invoke', [ApplyCouponToCart::createFromData('cart', 'couponCode')]);
     }
+
+    function it_removes_coupon_if_promotion_coupon_is_not_found(
+        OrderRepositoryInterface $orderRepository,
+        PromotionCouponRepositoryInterface $promotionCouponRepository,
+        OrderProcessorInterface $orderProcessor,
+        OrderInterface $cart
+    ): void {
+        $orderRepository->findCartByTokenValue('cart')->willReturn($cart);
+
+        $promotionCouponRepository->findOneBy(['code' => null])->willReturn(null);
```

Comment:
If I see correctly, it is not true. This method will not be used if the coupon code is null according to above implementation. 

This spec should be probably renamed to sth like `it_removes_coupon_if_passed_promotion_coupon_code_is_null`. 

There is also missing `$cart->setPromotionCoupon(null)->shouldBeCalled;`

Reply:
Fixed, also I added this missing step for the rest of specs